handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1260,10 +1277,12 @@ private SSLEngineResult sslReadErrorResult(int error, int stackError, int bytesC
         // This is needed so we ensure close_notify etc is correctly send to the remote peer.
         // See https://github.com/netty/netty/issues/3900
         if (SSL.bioLengthNonApplication(networkBIO) > 0) {
-            if (handshakeException == null && handshakeState != HandshakeState.FINISHED) {
+            if (pendingException == null) {
```

Comment:
I hardly know the control flow here but since this is not just `handshakeException` now, do we need to also handle a case when `pendingException != null`?

Reply:
it should never... let me add an assert tho